allennlp/common/from_params.py

```diff
@@ -334,14 +334,17 @@ def construct_arg(
 
     # If the parameter type is a Python primitive, just pop it off
     # using the correct casting pop_xyz operation.
-    elif annotation == str:
-        return popped_params
-    elif annotation == int:
-        return int(popped_params)  # type: ignore
-    elif annotation == bool:
-        return bool(popped_params)
+    elif annotation in {str, int, bool}:
+        if type(popped_params) in {str, int, bool}:
```

Comment:
Won't the popped param always be a float since it's coming from json? Highly suggest a test making it clear how this works.

Reply:
I checked it in the debugger just now. Ints come in as ints.

Reply:
```
In [14]: type(json.loads(json.dumps({'x':1}))['x'])
Out[14]: int
```

Reply:
Cool, thanks for correcting me! Still wouldn't mind that test just to prevent regressions.

Reply:
Added in 4462721.